File: Backend/SmartMicrogrid.API/Models/Transactions/Transaction.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace SmartMicrogrid.API.Models.Transactions
{
    public class Transaction
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = string.Empty;

        public string ReservationId { get; set; } = string.Empty;

        public string ProsumerId { get; set; } = string.Empty;

        public string MicrogridNodeId { get; set; } = string.Empty;

        public string EnergySlotId { get; set; } = string.Empty;

        public double EnergyAmount { get; set; }

        public string TransactionCode { get; set; } = string.Empty;

        public string QrCodeData { get; set; } = string.Empty;

        public string? VerifiedBy { get; set; }

        public DateTime? VerificationTime { get; set; }

        public DateTime? EnergyTransferTime { get; set; }

        public string Status { get; set; } = "Pending";

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Android/SmartMicrogrid.Android/app/src/main/java/com/smartmicrogrid/models/Transaction.kt
package com.smartmicrogrid.models

import com.google.gson.annotations.SerializedName

data class Transaction(
    @SerializedName("id")
    val id: String = "",

    @SerializedName("reservationId")
    val reservationId: String = "",

    @SerializedName("prosumerId")
    val prosumerId: String = "",

    @SerializedName("microgridNodeId")
    val microgridNodeId: String = "",

    @SerializedName("energySlotId")
    val energySlotId: String = "",

    @SerializedName("energyAmount")
    val energyAmount: Double = 0.0,

    @SerializedName("transactionCode")
    val transactionCode: String = "",

    @SerializedName("qrCodeData")
    val qrCodeData: String = "",

    @SerializedName("verifiedBy")
    val verifiedBy: String? = null,

    @SerializedName("verificationTime")
    val verificationTime: String? = null,

    @SerializedName("energyTransferTime")
    val energyTransferTime: String? = null,

    @SerializedName("status")
    val status: String = "Pending",

    @SerializedName("createdAt")
    val createdAt: String = "",

    @SerializedName("updatedAt")
    val updatedAt: String = ""
)